lib/pebbles/pebble_hierarchy.flow

```diff
@@ -238,18 +238,32 @@ setViewForPebbledHierarchyFilterParameters(
 	id : int,
 	parameters : [KeyValue]
 ) -> void {
-	allItems = getAllPebbledHierarchyItems(items);
-	index = findiDef(allItems, \item -> item.name == itemName, -1);
-	if (index >= 0 && index < length(allItems) - 1) {
-		item = allItems[index];
-
-		nextItem = allItems[index + 1];
-		setViewFromPebble(
-			controller,
-			makeSimplePebble3(nextItem.name, arrayPush(parameters, KeyValue(item.parameter, i2s(id))))
-		);
-		showMSnackbar(manager, formatString(_("Filter by %1 " + item.parameter + " activated"), [item.id2label(id)]), []);
-	}
+	itemM = findPebbleByName(items, itemName);
+	eitherFn(itemM, \item -> {
+		if (item.children != []) {
+			nextItem = item.children[0];
+			setViewFromPebble(
+				controller,
+				makeSimplePebble3(nextItem.name, arrayPush(parameters, KeyValue(item.parameter, i2s(id))))
+			);
+			showMSnackbar(manager, formatString(_("Filter by %1 " + item.parameter + " activated"), [item.id2label(id)]), []);
+		}
+	}, 
+	nop);
+}
+
+findPebbleByName(items : [PebbledHierarchyItem], itemName: string) -> Maybe<PebbledHierarchyItem> {
+	fold(items, None(), \acc, item -> {
+		if (item.name == itemName) {
+			Some(item)
+		} else {
+			findItemM = findPebbleByName(item.children, itemName);
```

Comment:
this looks excessive, should be enough for else branch:
```
else {
    findPebbleByName(item.children, itemName)
}
```
or even findmap instead of fold (I'm not sure what will be better here):
```
findmap(items, \item -> {
	if (item.name == itemName) {
		Some(item)
	} else {
		findPebbleByName(item.children, itemName))
	}
})
```